subprojects/core/src/main/java/org/gradle/initialization/layout/ProjectCacheDir.java

```diff
@@ -55,12 +57,16 @@ public void stop() {
             deleter,
             CleanupFrequency.DAILY
         );
-        String description = cleanupAction.getDisplayName();
-        ProgressLogger progressLogger = progressLoggerFactory.newOperation(ProjectCacheDir.class).start(description, description);
-        try {
-            cleanupAction.execute(new DefaultCleanupProgressMonitor(progressLogger));
-        } finally {
-            progressLogger.completed();
-        }
+        buildOperationRunner.run(new RunnableBuildOperation() {
+            @Override
+            public void run(BuildOperationContext context) {
+                cleanupAction.execute(new DefaultCleanupProgressMonitor(context));
+            }
+
+            @Override
+            public BuildOperationDescriptor.Builder description() {
+                return BuildOperationDescriptor.displayName(cleanupAction.getDisplayName());
+            }
+        });
```

Comment:
💭 And this is the third part of switching to using build operations instead of progress loggers.